src/jsifier.mjs

```diff
@@ -741,7 +761,13 @@ var proxiedFunctionTable = [
     finalCombiner();
   }
 
-  if (abortExecution) {
+  if (errorOccured()) {
     throw Error('Aborting compilation due to previous errors');
   }
 }
+
+addMacros({
+  extraLibraryFuncs,
+  addedLibraryItems,
+  preJS,
+});
```

Comment:
The name `addMacros` is confusing me. It makes me think of `C` macros - is that right or wrong? `addedLibraryItems` doesn't feel like a C macro so I'm guessing that's wrong somehow.

Reply:
Perhaps we can some up with a better name.

I think of the things inside of the `{{{` and `}}}` as macros because they get expanded to strings at built time.

Kind of the same thing for things in `#if` statements.  e.g. `#if FOO` in C means FOO is preprocessor macro.

I'm happy to consider better names if you can think of one?

Reply:
For `addMacros` perhaps `defineForMacros` - not the best name, but at least it clarifies it is defining something so that macros can use it, as opposed to adding macros themselves (which is what confused me)?

Or perhaps `macroContext => compileTimeContext` ; `addMacros => addToCompileTimeContext`? This stresses the compile-time aspect of it, which is what macros are an example of.

Reply:
I'm OK with `compileTimeContext` yes.

The reason I went with `addMacros` is that it is actually used to define macros that can then be expanded in library code.

e.g. 

We add `POINTER_SIZE` as a macro using `addMacros({POINTER_SIZE})` and then that macro is then available for use via:

```
{{{ POINTER_SIZE }}}
```

Here this macro expands to either 4 or 8 in the user's code.

When you say "it is defining something so that macros can use it" I'm not sure I understand.   If the function `foo()` is a macro that you add via `addMacros` and then you use it via `{{{ foo() }}}`, isn't that using it *as* a macro, not a macro using it.   I guess you mean that `foo` here is part of the inline `foo()` macro string maybe?

Reply:
Luckily `compileTimeContext` or `macroContext` won't be itself visible to macros/libraries so we can always rename it with impunity (thanks for this very change)!

Reply:
Yeah, good points, I sort of see what you mean now. So `POINTER_SIZE` is kind of a macro in that sense. What seems confusing is that it is just a JS variable in the compiler, so this sort of makes it "macro-able"...

I think it's clearer to avoid "macro" here for those reasons personally.

Reply:
(I also don't love `compileTimeContext` because all this happens at link time .. but I guess that `buildTimeContext` and `linkTimeContext` also don't roll off the tongue).